tests/test_mealie.py

```diff
@@ -172,6 +172,14 @@ async def response_handler(_: str, **_kwargs: Any) -> CallbackResult:
             assert await mealie_client.get_startup_info()
 
 
+async def test_client_connection_error() -> None:
+    """Test client connection error from mealie."""
+
+    async with MealieClient(api_host="https://bad-url") as mealie_client:
```

Comment:
Ah! I also made a fix on a local branch but couldn't figure out yet how its testable

Reply:
I struggled with this as well, but I'm learning 😄